astropy/io/fits/hdu/image.py

```diff
@@ -734,109 +733,80 @@ def __init__(self, hdu):
         self.hdu = hdu
 
     def __getitem__(self, key):
-        dims = []
         if not isinstance(key, tuple):
             key = (key,)
         naxis = len(self.hdu.shape)
-        if naxis < len(key):
-            raise IndexError('too many indices')
-        elif naxis > len(key):
-            key = key + (slice(None),) * (naxis - len(key))
+        return_scalar = (all(isinstance(k, (int, np.integer)) for k in key)
+                         and len(key) == naxis)
+        if not any(k is ... for k in key):
```

Comment:
This is invalid syntax on Python 2 (and below as well).  You need to use `Ellipsis`--the `...` literal does not work outside of slice notation.